File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution 
{
    public int[] productExceptSelf(int[] nums) 
    {
        if(nums.length == 0)
        {
            return nums;
        }

        int[] arr = new int[nums.length];
        int product;
        for(int i = 0; i < nums.length; i++)
        {
            product = 1;
            for(int j = 0; j < nums.length; j++)
            {
                if(j != i)
                {
                    product *= nums[j];
                }
            }

            arr[i] = product;
        }

        return arr;
    }
}  

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        if not nums:
            return []
        n = len(nums)
        res = []
        used = [False] * n 
        def backtrack(path):
            if len(path) == n:
                res.append(path[:])
                return

            for i in range(n):
                if used[i]:
                    continue
                path.append(nums[i])
                used[i] = True
                backtrack(path)
                used[i] = False
                path.pop()
        backtrack([])
        return res
